### DISCUSSION TITLE
How to debug tauri rust code with Rustrover?

### DISCUSSION THREAD

**Discussion by HuakunShen** [tauri-apps/tauri#8253] [Q&A]
JetBrains recently released Rustrover, an IDE for Rust. 
How to use its debugger to debug Tauri? 
A regular rust program that can be run with `cargo run` can be debugged easily. 

`cargo-tauri` is the CLI to run dev server and tauri core process.

I found this file

https://github.com/tauri-apps/tauri/blob/dev/tooling/cli/src/dev.rs

**Comment by HuakunShen** [comment]
Running `src/main.rs` directly works but not in dev mode, so the frontend must be built beforehand.

Not sure how to debug in dev mode.

**Comment by JonasKruckenberg** [comment]
Jetbrains products currently don't have support for background tasks so you will have to run the development server manually.
Then you should run the app as a regular rust program in the debugger.
 
See the CLion guide for details:
https://tauri.app/v1/guides/debugging/clion

**Reply by HuakunShen** [reply]
If I debug `src/main.rs`, it looks for frontend production build instead of localhost dev server. 
How to run `src/main.rs` directly but use dev server?
